ogr/services/pagure/project.py

```diff
@@ -545,3 +545,31 @@ def full_repo_name(self) -> str:
         fork = f"fork/{self._user}/" if self.is_fork else ""
         namespace = f"{self.namespace}/" if self.namespace else ""
         return f"{fork}{namespace}{self.repo}"
+
+    def __get_files(
+        self, path: str, ref: str = None, recursive: bool = False
+    ) -> Iterable[str]:
+        subfolders = ["."]
+
+        while subfolders:
+            path = subfolders.pop()
+            split_path = []
+            if path != ".":
+                split_path = ["f"] + path.split("/")
+            response = self._call_project_api("tree", ref, *split_path)
+
+            for file in response["content"]:
+                if file["type"] == "file":
+                    yield file["path"]
+                elif recursive and file["type"] == "folder":
+                    subfolders.append(file["path"])
+
+    def get_files(
+        self, ref: str = None, filter_regex: str = None, recursive: bool = False
+    ) -> List[str]:
```

Comment:
docstring here would be nice

Reply:
see #533, should be inherited from interface :)